<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1029</width>
    <height>643</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>0</y>
      <width>341</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>30</pointsize>
     </font>
    </property>
    <property name="text">
     <string>STM32 to LinuxQT</string>
    </property>
   </widget>
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>60</y>
      <width>921</width>
      <height>521</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>1</number>
    </property>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>TCP/IP</string>
     </attribute>
     <widget class="QDial" name="dial_servo">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>90</y>
        <width>81</width>
        <height>111</height>
       </rect>
      </property>
      <property name="notchesVisible">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_rxdata">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>300</y>
        <width>451</width>
        <height>161</height>
       </rect>
      </property>
      <property name="verticalScrollBarPolicy">
       <enum>Qt::ScrollBarAlwaysOn</enum>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_servo">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>140</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>80</x>
        <y>210</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>send data</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_send">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>240</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>send</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>360</x>
        <y>80</y>
        <width>121</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>DHT11 주기 조정</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spinBox_DHT11">
      <property name="geometry">
       <rect>
        <x>370</x>
        <y>50</y>
        <width>44</width>
        <height>26</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>10</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>80</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>현재온도</string>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_led">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>130</y>
        <width>64</width>
        <height>23</height>
       </rect>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_temp">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>20</y>
        <width>61</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="digitCount">
       <number>3</number>
      </property>
     </widget>
     <widget class="QLCDNumber" name="lcdNumber_wet">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>20</y>
        <width>61</width>
        <height>51</height>
       </rect>
      </property>
      <property name="font">
       <font>
        <pointsize>20</pointsize>
       </font>
      </property>
      <property name="digitCount">
       <number>3</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>270</y>
        <width>111</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>STM32 rxData</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>170</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>서보모터 PWM</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="lineEdit_send">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>240</y>
        <width>341</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string/>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>80</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>현재습도</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_clear">
      <property name="geometry">
       <rect>
        <x>440</x>
        <y>150</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Clear</string>
      </property>
     </widget>
     <widget class="QDial" name="dial_led">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>80</y>
        <width>81</width>
        <height>111</height>
       </rect>
      </property>
      <property name="notchesVisible">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>160</y>
        <width>67</width>
        <height>17</height>
       </rect>
      </property>
      <property name="text">
       <string>LED PWM</string>
      </property>
     </widget>
     <widget class="QCustomPlot" name="plot" native="true">
      <property name="geometry">
       <rect>
        <x>560</x>
        <y>70</y>
        <width>331</width>
        <height>381</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_2">
     <attribute name="title">
      <string>UART</string>
     </attribute>
     <widget class="QGroupBox" name="groupBox">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>40</y>
        <width>271</width>
        <height>341</height>
       </rect>
      </property>
      <property name="title">
       <string>PortConfig</string>
      </property>
      <widget class="QLabel" name="label_9">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>50</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Device</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_10">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>90</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>BuadRate</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_11">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>130</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>DataBits</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_12">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>170</y>
         <width>67</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>Parity Bits</string>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_Device">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>50</y>
         <width>86</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_BuadRate">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>90</y>
         <width>86</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_DataBits">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>130</y>
         <width>86</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QComboBox" name="comboBox_ParityBits">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>170</y>
         <width>86</width>
         <height>25</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_OpenPort">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>260</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>OpenPort</string>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_ClosePort">
       <property name="geometry">
        <rect>
         <x>140</x>
         <y>260</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>ClosePort</string>
       </property>
      </widget>
      <widget class="QLabel" name="label_13">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>220</y>
         <width>81</width>
         <height>17</height>
        </rect>
       </property>
       <property name="text">
        <string>PortStatus</string>
       </property>
      </widget>
      <widget class="QProgressBar" name="progressBar">
       <property name="geometry">
        <rect>
         <x>130</x>
         <y>220</y>
         <width>118</width>
         <height>23</height>
        </rect>
       </property>
       <property name="value">
        <number>0</number>
       </property>
       <property name="textVisible">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_ScanAgain">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>300</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>ScanAgain</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>50</y>
        <width>501</width>
        <height>91</height>
       </rect>
      </property>
      <property name="title">
       <string>Send</string>
      </property>
      <widget class="QLineEdit" name="lineEdit_SendUart">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>40</y>
         <width>381</width>
         <height>31</height>
        </rect>
       </property>
      </widget>
      <widget class="QPushButton" name="pushButton_SendUart">
       <property name="geometry">
        <rect>
         <x>400</x>
         <y>50</y>
         <width>89</width>
         <height>25</height>
        </rect>
       </property>
       <property name="text">
        <string>Send</string>
       </property>
      </widget>
     </widget>
     <widget class="QGroupBox" name="groupBox_3">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>160</y>
        <width>511</width>
        <height>231</height>
       </rect>
      </property>
      <property name="title">
       <string>Recv</string>
      </property>
      <widget class="QTextEdit" name="textEdit_RecvUart">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>30</y>
         <width>481</width>
         <height>191</height>
        </rect>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOn</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOn</enum>
       </property>
      </widget>
     </widget>
     <widget class="QPushButton" name="pushButton_Clear">
      <property name="geometry">
       <rect>
        <x>770</x>
        <y>400</y>
        <width>89</width>
        <height>25</height>
       </rect>
      </property>
      <property name="text">
       <string>Clear</string>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar_LED1">
      <property name="geometry">
       <rect>
        <x>60</x>
        <y>390</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar_LED2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>390</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QProgressBar" name="progressBar_LED3">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>390</y>
        <width>71</width>
        <height>21</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
      <property name="textVisible">
       <bool>false</bool>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_LED1">
      <property name="geometry">
       <rect>
        <x>50</x>
        <y>430</y>
        <width>92</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>LED1</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_LED2">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>430</y>
        <width>92</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>LED2</string>
      </property>
     </widget>
     <widget class="QCheckBox" name="checkBox_LED3">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>430</y>
        <width>92</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>LED3</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1029</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QCustomPlot</class>
   <extends>QWidget</extends>
   <header>qcustomplot.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
